common/src/main/java/io/homeassistant/companion/android/common/data/keychain/KeyStoreImpl.kt

```diff
@@ -0,0 +1,103 @@
+package io.homeassistant.companion.android.common.data.keychain
+
+import android.content.Context
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.security.KeyStore
+import java.security.KeyStore.PrivateKeyEntry
+import java.security.PrivateKey
+import java.security.cert.X509Certificate
+import javax.inject.Inject
+
+class KeyStoreImpl @Inject constructor() : KeyChainRepository {
+    companion object {
+        private const val TAG = "KeyStoreRepository"
+        const val ALIAS = "TLSClientCertificate"
+    }
+
+    private var alias: String? = null
+    private var key: PrivateKey? = null
+    private var chain: Array<X509Certificate>? = null
+
+    override suspend fun clear() {
+        alias = null
+        key = null
+        chain = null
```

Comment:
While not explicitly documented, looking at the existing implementation this function in the interface is supposed to permanently clear the certificate settings known/available to the app, not just the variables.

To prevent future issues, I suggest moving this code to another (private) function.